diff --git a/RL_Algorithm/DT_ECO_mosac_dicrete_action.py b/RL_Algorithm/DT_ECO_mosac_dicrete_action.py
index 472cee0..0a7182f 100644
--- a/RL_Algorithm/DT_ECO_mosac_dicrete_action.py
+++ b/RL_Algorithm/DT_ECO_mosac_dicrete_action.py
@@ -536,7 +536,7 @@ class MOSAC(MOPolicy, MOAgent):
             if actions.ndim == 2:
                 actions = np.squeeze(actions)
             next_obs, rewards, terminated, truncated, infos = self.env.step(actions)
-            print(rewards)
+            print(f'Timing Step: {step}, rewards: {rewards}')
 
             # TRY NOT TO MODIFY: save data to reply buffer; handle `final_observation`
             real_next_obs = next_obs
diff --git a/dt_eco/dt_eco.py b/dt_eco/dt_eco.py
index 46580e2..08a29d6 100644
--- a/dt_eco/dt_eco.py
+++ b/dt_eco/dt_eco.py
@@ -31,7 +31,6 @@ search_paths = [
 module_dir = find_module_dir('Global_var', search_paths)
 if module_dir:
     sys.path.append(module_dir)
-    print(f"Project Main dir found by the 'dt_eco' environment file.")
 else:
     raise ImportError("Project Main dir not found by the 'dt_eco' environment file. Please check the working directory!")
 import Global_var
@@ -271,7 +270,7 @@ class DT_ECO(gym.Env):
                     Interaction.VerilogInlineChange(self.current_design, changed_cell_dict, Incremental=True) 
                     
                 # # run pt
-                Interaction.VerilogInline_PT_Iteration(self.current_design)
+                # Interaction.VerilogInline_PT_Iteration(self.current_design)
                 # Timing Graph update (inline)
                 self.graph = TimingGraphTrans.LoadTimingGraph(self.current_design+'_inline', True)               
                 self.nodes, self.nodes_rev = TimingGraphTrans.LoadNodeDict(self.current_design+'_inline')
@@ -328,7 +327,7 @@ class DT_ECO(gym.Env):
         return self._get_obs(done), vec_reward, done, False, info
 
     def _get_gate_sizes(self):
-        gate_sizes = np.zeros(len(self.CriticalPaths[self.current_path_index].Cellname_to_Cell.keys()), dtype=np.int32)  # total number of gates on paths
+        gate_sizes = np.zeros(len(self.CriticalPaths[self.current_path_index].Cellname_to_Cell.keys()), dtype=np.float32)  # total number of gates on paths
         gate_sizes[:len(self.chosen_sizes)] = [val + self.gate_size_list[1] for val in self.chosen_sizes] # plus 1/max_length to avoid 0, since 0 means gate remain its original size
         self.gate_sizes[self.current_path_index, :len(gate_sizes)] = gate_sizes # update the gate sizes observation
         return gate_sizes
@@ -342,27 +341,27 @@ class DT_ECO(gym.Env):
             if chosedsize == 0:
                 pass # 0 means remain the original size
             elif chosedsize <= maxsize:
-                chosedsize = self.gate_size_list.index(chosedsize - self.gate_size_list[1]) # resore the size to the index
+                # restore the key value to the action space, consider the distubance introduced by minus and plus
+                chosedsize = min(range(len(self.gate_size_list)), key=lambda i: abs(self.gate_size_list[i] - (chosedsize - self.gate_size_list[1])))
                 newsize = self.footprint[cellfootprint][chosedsize]
                 sizedCell[cellname] = [originalsize, newsize]
             else:
                 pass # invalid size
+        print(sizedCell)    
         return sizedCell
 
 if __name__ == "__main__": 
     start_time = time.time()
     env = DT_ECO()
     env.reset()
-    # for _ in range(env.steps_needed):
-    #     env.render()
-    #     action = env.action_space.sample() # random action space sampling
-    #     # print(f'action: {action}#{type(action)}')
-    #     current_state, vec_reward, done, truncated, info = env.step(action)
-    #     # np.set_printoptions(edgeitems=10, threshold=20, precision=4, suppress=False, linewidth=np.inf)
-    #     # print(current_state['gate_sizes'])
-    #     print(f'reward: {vec_reward}')
-    #     if done:
-    #         env.reset()
+    for i in range(env.steps_needed):
+        env.render()
+        action = env.action_space.sample() # random action space sampling
+        current_state, vec_reward, done, truncated, info = env.step(action)
+        # np.set_printoptions(edgeitems=10, threshold=20, precision=4, suppress=False, linewidth=np.inf)
+        print(f'step {i} reward: {vec_reward}')
+        if done:
+            env.reset()
     env.close()
     end_time = time.time()
     elapsed_time = end_time - start_time
diff --git a/work/VerilogInline/aes_cipher_top_route.v b/work/VerilogInline/aes_cipher_top_route.v
index 65b93b0..8543409 100644
--- a/work/VerilogInline/aes_cipher_top_route.v
+++ b/work/VerilogInline/aes_cipher_top_route.v
@@ -16145,7 +16145,7 @@ DQV4_90S9T16R sa11_reg_3_ ( .D ( N197 ) , .CK ( ctsbuf_net_34472 ) ,
     .VSS ( VSS ) ) ;
 CLKNAND2V3_90S9T16R U862 ( .A1 ( n1064 ) , .A2 ( n1065 ) , .ZN ( n603 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R U863 ( .A1 ( n1064 ) , .A2 ( n1065 ) , .ZN ( n1268 ) , 
+NAND2V1_90S9T16R U863 ( .A1 ( n1064 ) , .A2 ( n1065 ) , .ZN ( n1268 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKNAND2V2_90S9T16R U864 ( .A1 ( n1400 ) , .A2 ( n1063 ) , .ZN ( n1064 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -16748,7 +16748,7 @@ INV2_90S9T16R U1155 ( .I ( sa12_sr[6] ) , .ZN ( n1506 ) , .VDD ( VDD ) ,
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND2V2_90S9T16R U1156 ( .A1 ( n1035 ) , .A2 ( n1036 ) , .ZN ( n1548 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R U1157 ( .A1 ( n1217 ) , .A2 ( n988 ) , .ZN ( n1035 ) , 
+CLKNAND2V2_90S9T16R U1157 ( .A1 ( n1217 ) , .A2 ( n988 ) , .ZN ( n1035 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 XOR3V1_90S9T16R U1158 ( .A1 ( w1[14] ) , .A2 ( sa31_sr[6] ) , 
     .A3 ( sa31_sr[5] ) , .Z ( n1133 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
@@ -16946,7 +16946,7 @@ NAND2V2_90S9T16R U1257 ( .A1 ( n1060 ) , .A2 ( n1061 ) , .ZN ( N69 ) ,
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKNAND2V1_90S9T16R U1258 ( .A1 ( n1314 ) , .A2 ( sa01_sr[2] ) , 
     .ZN ( n703 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R U1259 ( .A1 ( n701 ) , .A2 ( n702 ) , .ZN ( n704 ) , 
+CLKNAND2V1_90S9T16R U1259 ( .A1 ( n701 ) , .A2 ( n702 ) , .ZN ( n704 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKNAND2V4_90S9T16R U1260 ( .A1 ( n703 ) , .A2 ( n704 ) , .ZN ( n1081 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -17170,7 +17170,7 @@ INV2_90S9T16R ZINV_4_inst_1674 ( .I ( n5155 ) , .ZN ( ZINV_4_91 ) ,
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV1_90S9T16R U1378 ( .I ( n698 ) , .ZN ( n755 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-INV4_90S9T16R U1379 ( .I ( n1243 ) , .ZN ( n1453 ) , .VDD ( VDD ) , 
+INV8_90S9T16R U1379 ( .I ( n1243 ) , .ZN ( n1453 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV1_90S9T16R U1380 ( .I ( n1288 ) , .ZN ( n757 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -17824,7 +17824,7 @@ XNOR2V5_90S9T16R U1705 ( .A1 ( n1482 ) , .A2 ( sa22_sr[7] ) , .ZN ( n1074 ) ,
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 XOR2V3_90S9T16R U1706 ( .A1 ( w2[15] ) , .A2 ( n962 ) , .Z ( N458 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-INV8_90S9T16R U1707 ( .I ( sa01_sr[7] ) , .ZN ( n1314 ) , .VDD ( VDD ) , 
+INV4_90S9T16R U1707 ( .I ( sa01_sr[7] ) , .ZN ( n1314 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV1_90S9T16R U1709 ( .I ( n672 ) , .ZN ( n1011 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -17943,7 +17943,7 @@ XOR2V5_90S9T16R U1758 ( .A1 ( n124 ) , .A2 ( n269 ) , .Z ( n1349 ) ,
 NOR2V4_90S9T16R ctmTdsLR_2_9024 ( .A1 ( n4999 ) , .A2 ( n50010 ) , 
     .ZN ( popt_net_3841 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-INV8_90S9T16R U1760 ( .I ( sa33_sr[7] ) , .ZN ( net58265 ) , .VDD ( VDD ) , 
+INV4_90S9T16R U1760 ( .I ( sa33_sr[7] ) , .ZN ( net58265 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND2V2_90S9T16R U1761 ( .A1 ( sa12_sr[2] ) , .A2 ( ZINV_141_175 ) , 
     .ZN ( n1051 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
@@ -19237,7 +19237,7 @@ AOI21V1_90S9T16R ctmTdsLR_4_3529 ( .A1 ( n7039 ) , .A2 ( n49814 ) ,
 CLKNAND2V4_90S9T16R ctmTdsLR_1_5536 ( .A1 ( n2492 ) , .A2 ( n2252 ) , 
     .ZN ( n15511 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R ctmTdsLR_1_5537 ( .A1 ( popt_net_2129 ) , 
+CLKNAND2V4_90S9T16R ctmTdsLR_1_5537 ( .A1 ( popt_net_2129 ) , 
     .A2 ( popt_net_2131 ) , .ZN ( n2969_CDR1 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKNAND2V2_90S9T16R ctmTdsLR_1_5695 ( .A1 ( popt_net_2227 ) , 
@@ -19595,7 +19595,7 @@ XNOR2V5_90S9T16R U139 ( .A1 ( w0[16] ) , .A2 ( ZBUF_51_20 ) , .ZN ( n286 ) ,
 NAND3BBV2_90S9T16R ctmTdsLR_16_2804 ( .A1 ( n1558 ) , .A2 ( sa00_sr[4] ) , 
     .B ( popt_net_567 ) , .ZN ( popt_net_568 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R ctmTdsLR_4_3917 ( .A1 ( n4094 ) , .A2 ( n50112 ) , 
+CLKNAND2V1_90S9T16R ctmTdsLR_4_3917 ( .A1 ( n4094 ) , .A2 ( n50112 ) , 
     .ZN ( popt_net_1230 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 OAI22BBV3_90S9T16R U136 ( .B1 ( HFSNET_15 ) , .B2 ( ZINV_4_191 ) , 
@@ -23013,7 +23013,7 @@ NOR3V0P5_90S9T16R U3033 ( .A1 ( n5342 ) , .A2 ( n6833 ) , .A3 ( n5332_CDR1 ) ,
 OAOI211V2_90S9T16R U3023 ( .A1 ( n8353 ) , .A2 ( n8323 ) , .B ( n8662 ) , 
     .C ( n5552_CDR1 ) , .ZN ( n5562 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNOR2V8_90S9T16R ctmTdsLR_9_6183 ( .A1 ( popt_net_2495 ) , .A2 ( n858 ) , 
+CLKNOR2V2_90S9T16R ctmTdsLR_9_6183 ( .A1 ( popt_net_2495 ) , .A2 ( n858 ) , 
     .ZN ( popt_net_2496 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 NOR4V2_90S9T16R U3004 ( .A1 ( n5432_CDR1 ) , .A2 ( n5422_CDR1 ) , 
@@ -25547,7 +25547,7 @@ CLKINV10_90S9T16R U2675 ( .I ( n2312 ) , .ZN ( n2761 ) , .VDD ( VDD ) ,
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND2V1_90S9T16R U2665 ( .A1 ( n4464_CDR1 ) , .A2 ( n3411 ) , .ZN ( n8215 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R U2655 ( .A1 ( n8215 ) , .A2 ( n8116 ) , .ZN ( n2821 ) , 
+CLKNAND2V2_90S9T16R U2655 ( .A1 ( n8215 ) , .A2 ( n8116 ) , .ZN ( n2821 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 AOI22V1_90S9T16R U2645 ( .A1 ( n4334_CDR1 ) , .A2 ( n4974_CDR1 ) , 
     .B1 ( n4354_CDR1 ) , .B2 ( ZBUF_28_31 ) , .ZN ( n10911 ) , .VDD ( VDD ) , 
@@ -26353,7 +26353,7 @@ AND3V4_90S9T16R U4156 ( .A1 ( net56654 ) , .A2 ( net56734 ) , .A3 ( n2100 ) ,
 NOR2XBV3_90S9T16R ctmTdsLR_5_4084 ( .A1 ( popt_net_1341 ) , .B1 ( n1191 ) , 
     .ZN ( popt_net_1342 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R ctmTdsLR_3_5287 ( .A1 ( n1382 ) , .A2 ( popt_net_1977 ) , 
+NAND2V1_90S9T16R ctmTdsLR_3_5287 ( .A1 ( n1382 ) , .A2 ( popt_net_1977 ) , 
     .ZN ( popt_net_1978 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 NOR2V0P5_90S9T16R U4126 ( .A1 ( n2922 ) , .A2 ( n1583 ) , .ZN ( n2972 ) , 
@@ -27064,7 +27064,7 @@ NOR2V0P5_90S9T16R U14515 ( .A1 ( n3722 ) , .A2 ( n3331 ) ,
 NAND4BBV4_90S9T16R ctmTdsLR_1_8271 ( .A1 ( n2504 ) , .A2 ( ZBUF_90_168 ) , 
     .B1 ( n5218 ) , .B2 ( n5317 ) , .ZN ( n5404 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-INV8_90S9T16R U14315 ( .I ( net56754_CDR1 ) , .ZN ( n4316 ) , .VDD ( VDD ) , 
+INV4_90S9T16R U14315 ( .I ( net56754_CDR1 ) , .ZN ( n4316 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV1_90S9T16R U14215 ( .I ( n3252 ) , .ZN ( n5315 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -28127,7 +28127,7 @@ NAND4BBV4_90S9T16R U2307 ( .A1 ( n3842_CDR1 ) , .A2 ( n4214_CDR1 ) ,
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 AND2V6_90S9T16R U2297 ( .A1 ( sa02[6] ) , .A2 ( n15712 ) , .Z ( n15113 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-INV8_90S9T16R U2287 ( .I ( sa02[7] ) , .ZN ( n5018 ) , .VDD ( VDD ) , 
+INV2_90S9T16R U2287 ( .I ( sa02[7] ) , .ZN ( n5018 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NOR2XBV8_90S9T16R U2277 ( .A1 ( sa02[6] ) , .B1 ( n5018 ) , .ZN ( n14513 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -31916,7 +31916,7 @@ NAND4BBV4_90S9T16R U21919 ( .A1 ( n21111 ) , .A2 ( n2126 ) , .B1 ( n7125 ) ,
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV2_90S9T16R U21819 ( .I ( n3795 ) , .ZN ( n6920 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R U21719 ( .A1 ( n1766 ) , .A2 ( n1706 ) , .ZN ( n5078_CDR1 ) , 
+NAND2V6_90S9T16R U21719 ( .A1 ( n1766 ) , .A2 ( n1706 ) , .ZN ( n5078_CDR1 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NOR3V1_90S9T16R U21619 ( .A1 ( n46210_CDR1 ) , .A2 ( n3376 ) , .A3 ( n3585 ) , 
     .ZN ( n2097 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
@@ -32112,7 +32112,7 @@ AND2V2_90S9T16R U12919 ( .A1 ( n46210_CDR1 ) , .A2 ( n4508_CDR1 ) ,
 NAND2V2_90S9T16R U12819 ( .A1 ( n5048_CDR1 ) , .A2 ( n4508_CDR1 ) , 
     .ZN ( n2476 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R U12719 ( .A1 ( n4508_CDR1 ) , .A2 ( n4938_CDR1 ) , 
+NAND2V3_90S9T16R U12719 ( .A1 ( n4508_CDR1 ) , .A2 ( n4938_CDR1 ) , 
     .ZN ( n2827 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 OAI221V1_90S9T16R U12619 ( .A1 ( n4678 ) , .A2 ( n11106 ) , .B1 ( n7719 ) , 
@@ -33238,7 +33238,7 @@ NAND3V2_90S9T16R U15520 ( .A1 ( n3596 ) , .A2 ( ZBUF_12_161 ) ,
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
 INV2_90S9T16R U15420 ( .I ( n3926 ) , .ZN ( n2828 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNOR2V4_90S9T16R ctmTdsLR_1_8141 ( .A1 ( n5047_CDR1 ) , 
+CLKNOR2V8_90S9T16R ctmTdsLR_1_8141 ( .A1 ( n5047_CDR1 ) , 
     .A2 ( popt_net_3516 ) , .ZN ( popt_net_1590 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND2V6_90S9T16R U15220 ( .A1 ( n9916 ) , .A2 ( n10616 ) , .ZN ( n4709 ) , 
@@ -33472,7 +33472,7 @@ AO22V1_90S9T16R U4720 ( .A1 ( n4759_CDR1 ) , .A2 ( n4699_CDR1 ) ,
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 AND2V10_90S9T16R U4627 ( .A1 ( n11516 ) , .A2 ( n11417 ) , .Z ( n10016 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
-INV8_90S9T16R U4540 ( .I ( sa12[4] ) , .ZN ( n3129 ) , .VDD ( VDD ) , 
+INV0P5_90S9T16R U4540 ( .I ( sa12[4] ) , .ZN ( n3129 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKNAND2V1_90S9T16R ctmTdsLR_2_6073 ( .A1 ( n4219 ) , .A2 ( n4063 ) , 
     .ZN ( popt_net_2442 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
@@ -33485,7 +33485,7 @@ NAND3V2_90S9T16R U4170 ( .A1 ( n4239 ) , .A2 ( n3346 ) , .A3 ( n4339 ) ,
 NAND2XBV2_90S9T16R ctmTdsLR_4_3750 ( .A1 ( n3743 ) , .B1 ( n3724 ) , 
     .ZN ( popt_net_1115 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R U3950 ( .A1 ( n4339 ) , .A2 ( n4239 ) , .ZN ( n1857 ) , 
+CLKNAND2V3_90S9T16R U3950 ( .A1 ( n4339 ) , .A2 ( n4239 ) , .ZN ( n1857 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 CLKINV12_90S9T16R U3850 ( .I ( n4339 ) , .ZN ( n4699_CDR1 ) , .VDD ( VDD ) , 
     .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
@@ -40784,7 +40784,7 @@ CLKNOR2V1_90S9T16R U28219 ( .A1 ( n18215_CDR1 ) , .A2 ( n18413_CDR1 ) ,
 NOR2XBV1_90S9T16R ctmTdsLR_6_2853 ( .A1 ( n1085 ) , .B1 ( popt_net_600 ) , 
     .ZN ( popt_net_601 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R ctmTdsLR_7_2854 ( .A1 ( n1090 ) , .A2 ( popt_net_599 ) , 
+NAND2V2_90S9T16R ctmTdsLR_7_2854 ( .A1 ( n1090 ) , .A2 ( popt_net_599 ) , 
     .ZN ( popt_net_600 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 INV2_90S9T16R U27917 ( .I ( n5080 ) , .ZN ( n6535 ) , .VDD ( VDD ) , 
@@ -42302,7 +42302,7 @@ INV3_90S9T16R U21027 ( .I ( net50815_CDR1 ) , .ZN ( n2380 ) , .VDD ( VDD ) ,
 CLKNAND2V2_90S9T16R U20927 ( .A1 ( net50865 ) , .A2 ( n36814_CDR1 ) , 
     .ZN ( n21614 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R U20727 ( .A1 ( net61702 ) , .A2 ( n15224 ) , .ZN ( n36215 ) , 
+NAND2V6_90S9T16R U20727 ( .A1 ( net61702 ) , .A2 ( n15224 ) , .ZN ( n36215 ) , 
     .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND2V8_90S9T16R U20627 ( .A1 ( n15124 ) , .A2 ( net62053 ) , 
     .ZN ( net50913 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
@@ -43341,7 +43341,7 @@ NAND3BV2_90S9T16R ctmTdsLR_3_6858 ( .A1 ( popt_net_2069 ) , .B1 ( n1805 ) ,
 OR4V4_90S9T16R U34221 ( .A1 ( n16715 ) , .A2 ( n16615 ) , .A3 ( n16515 ) , 
     .A4 ( n16415 ) , .Z ( n35914 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R ctmTdsLR_3_2649 ( .A1 ( n1519 ) , .A2 ( popt_net_460 ) , 
+CLKNAND2V2_90S9T16R ctmTdsLR_3_2649 ( .A1 ( n1519 ) , .A2 ( popt_net_460 ) , 
     .ZN ( popt_net_461 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 CLKNOR2V0P5_90S9T16R ctmTdsLR_3_9636 ( .A1 ( n8112 ) , .A2 ( n583 ) , 
@@ -45731,7 +45731,7 @@ CLKNAND2V2_90S9T16R ctmTdsLR_3_1949 ( .A1 ( n4460 ) , .A2 ( n4360 ) ,
 NOR2V1_90S9T16R ctmTdsLR_4_1950 ( .A1 ( n17510_CDR1 ) , .A2 ( n17410_CDR1 ) , 
     .ZN ( popt_net_19 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R ctmTdsLR_1_1951 ( .A1 ( popt_net_21 ) , 
+CLKNAND2V10_90S9T16R ctmTdsLR_1_1951 ( .A1 ( popt_net_21 ) , 
     .A2 ( popt_net_24 ) , .ZN ( sa01_sr[5] ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NOR4V1_90S9T16R ctmTdsLR_2_7040 ( .A1 ( n2281_CDR1 ) , .A2 ( n11510_CDR1 ) , 
@@ -47619,7 +47619,7 @@ NOR4BBV4_90S9T16R ctmTdsLR_4_3375 ( .A1 ( n8725 ) , .A2 ( n8537 ) ,
 NAND3BBV4_90S9T16R ctmTdsLR_1_3496 ( .A1 ( n51014 ) , .A2 ( popt_net_954 ) , 
     .B ( n4149 ) , .ZN ( sa20_sr[2] ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
-CLKNAND2V8_90S9T16R ctmTdsLR_1_3377 ( .A1 ( popt_net_908 ) , 
+CLKNAND2V12_90S9T16R ctmTdsLR_1_3377 ( .A1 ( popt_net_908 ) , 
     .A2 ( popt_net_738 ) , .ZN ( sa33_sr[5] ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
 NAND3V6_90S9T16R ctmTdsLR_1_9532 ( .A1 ( n41212 ) , .A2 ( n41114 ) , 
@@ -50447,7 +50447,7 @@ CLKINV0P5_90S9T16R ctmTdsLR_5_6248 ( .I ( popt_net_2527 ) ,
 NAND2XBV0P5_90S9T16R ctmTdsLR_6_6249 ( .A1 ( popt_net_1854 ) , 
     .B1 ( HFSNET_18 ) , .ZN ( popt_net_2527 ) , .VDD ( VDD ) , .VPW ( VSS ) , 
     .VNW ( VDD ) , .VSS ( VSS ) ) ;
-NAND2V8_90S9T16R ctmTdsLR_1_6309 ( .A1 ( n678 ) , .A2 ( n677 ) , 
+NAND2V4_90S9T16R ctmTdsLR_1_6309 ( .A1 ( n678 ) , .A2 ( n677 ) , 
     .ZN ( n1204 ) , .VDD ( VDD ) , .VPW ( VSS ) , .VNW ( VDD ) , 
     .VSS ( VSS ) ) ;
 AO22V1_90S9T16R ctmTdsLR_1_6251 ( .A1 ( HFSNET_20 ) , .A2 ( key[27] ) , 
